website/src/components/Chat/PluginsChooser.tsx

```diff
@@ -0,0 +1,257 @@
+import {
+  Button,
+  Text,
+  Tooltip,
+  Radio,
+  RadioGroup,
+  Box,
+  Editable,
+  EditableInput,
+  EditablePreview,
+  Image,
+  FormControl,
+  FormLabel,
+  Icon,
+  IconButton,
+  Menu,
+  MenuButton,
+  MenuItem,
+  MenuList,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Stack,
+  Textarea,
+  useDisclosure,
+} from "@chakra-ui/react";
+import { Cog, Edit, Plus } from "lucide-react";
+// i need
+import { AttachmentIcon, WarningIcon, CheckCircleIcon, CloseIcon } from "@chakra-ui/icons";
+import { ChangeEvent, useCallback, useEffect, useState, useRef } from "react";
+import { useTranslation } from "next-i18next";
+import { Controller, useFormContext } from "react-hook-form";
+import { ChatConfigFormData } from "src/types/Chat";
+import { useChatContext } from "./ChatContext";
+import { PluginEntry } from "src/types/Chat";
+import { get, post } from "src/lib/api";
+import { API_ROUTES } from "src/lib/routes";
+
+// TODO: Delete me!
+const measureDivWidth = (div: HTMLDivElement | null) => {
+  if (!div) return 0;
+  const width = div?.parentElement?.offsetWidth - 15;
+  return width;
+};
+
+export const PluginsChooser = ({ plugins }: { plugins: PluginEntry[] }) => {
+  const { t } = useTranslation("common");
+  const { control, register, reset, setValue } = useFormContext<ChatConfigFormData>();
+  const [selectedPluginIndex, setSelectedPluginIndex] = useState<number | null>(null);
+  const { isOpen, onOpen, onClose } = useDisclosure();
+
+  const textareaRef = useRef<HTMLTextAreaElement | null>(null);
+  const containerRef = useRef<HTMLDivElement | null>(null);
+
+  const findSelectedPluginIndex = useCallback(() => {
+    return plugins.findIndex((plugin) => plugin.enabled);
+  }, [plugins]);
+
+  const handlePluginSelect = useCallback(
+    (index: number) => {
+      const selectedIndex = plugins.findIndex((plugin) => plugin.enabled);
+      const newPlugins = plugins.map((plugin, i) => ({
+        ...plugin,
+        enabled: selectedIndex === index ? false : i === index,
+      }));
+      setValue("plugins", newPlugins, { shouldDirty: true });
+      // onPluginActivated(newPlugins[index].enabled);
+    },
+    [setValue, plugins]
+  );
+
+  const handlePluginEdit = useCallback(
+    (index: number) => {
+      setSelectedPluginIndex(index);
+      onOpen();
+    },
+    [onOpen]
+  );
+
+  const handlePluginRemove = useCallback(
+    (index) => {
+      const updatedPlugins = plugins.filter((_, idx) => idx !== index);
+      setValue("plugins", updatedPlugins, { shouldDirty: true });
+    },
+    [setValue, plugins]
+  );
+
+  const handlePluginSave = useCallback(async () => {
+    if (textareaRef.current) {
+      const newPlugins = [...plugins];
+      newPlugins[selectedPluginIndex!].url = textareaRef.current.value;
+
+      const plugin: PluginEntry = await post(API_ROUTES.GET_PLUGIN_CONFIG, {
```

Comment:
can't we fetch the plugin config initially once?

Reply:
How do you mean once?

Reply:
we already fetch all of the plugins from the backend, why not deliver the configuration immediately? why is it a separate request for the config? what is the added benefit of separating them.

Reply:
No you didn't get that correctly I think ?
When user adds new plugin, than we need to get all rest of the data from the backend, because of CORS issues ofc.

Initial fetching from the backend is only for OA out-of-the-box plugins. 
All other plugins you can add, and remove when ever you want, it's all described I think in pr and plugins.md
But its really just same as ChatGPT plugin system from the UI pov.

Reply:
btw, there will be no OA out-of-the-box plugins for this PR, we will be adding those separately, as I wrote on discord,
calculator plugin will be deleted, in this pr.